lib/api/BackbeatServer.js

```diff
@@ -0,0 +1,233 @@
+'use strict'; // eslint-disable-line
+
+const http = require('http');
+const os = require('os');
+
+const { Clustering, ipCheck } = require('arsenal');
+const werelogs = require('werelogs');
+
+const Config = require('../../conf/config.json');
+const BackbeatRequest = require('./BackbeatRequest');
+const BackbeatAPI = require('./BackbeatAPI');
+
+const Logger = werelogs.Logger;
+
+werelogs.configure({
+    level: Config.log.logLevel,
+    dump: Config.log.dumpLevel,
+});
+
+class BackbeatServer {
+    /**
+     * @constructor
+     * @param {Worker} [worker=null] - Track the worker using cluster
+     * @param {object} config - configurations for server setup
+     * @param {werelogs.Logger} logger - Logger object
+     * @param {BackbeatAPI} backbeatAPI - BackbeatAPI instance
+     */
+    constructor(worker, config, logger, backbeatAPI) {
+        this.server = null;
+        this.worker = worker;
+        this._config = config;
+        this._port = config.port;
+        this._maxISR = config.replicas;
+        this._logger = logger;
+        this.backbeatAPI = backbeatAPI;
+    }
+
+    /**
+     * Last log message to send per incoming request
+     * @param {werelogs.newRequestLogger} logger - request logger
+     * @param {object} req - request object
+     * @param {object} res - response object
+     * @return {undefined}
+     *
+     * @static
+     */
+    static logRequestEnd(logger, req, res) {
+        const info = {
+            clientIp: req.socket.remoteAddress,
+            clientPort: req.socket.remotePort,
+            httpMethod: req.method,
+            httpURL: req.url,
+            httpCode: res.statusCode,
+            httpMessage: res.statusMessage,
+        };
+        logger.end('finished handling request', info);
+    }
+
+    /**
+     * Check if incoming request is valid
+     * @param {object} req - incoming request object
+     * @param {BackbeatRequest} backbeatRequest - Backbeat request object
+     * @return {boolean} true if no errors
+     */
+    _isValidRequest(req, backbeatRequest) {
+        const allowIp = ipCheck.ipMatchCidrList(
+            this._config.healthChecks.allowFrom, req.socket.remoteAddress);
+        if (!allowIp) {
+            backbeatRequest.setError({
```

Comment:
You should set this to an arsenal error object instead (from `require('arsenal').errors`).